resources/assets/coffee/react/mp-history/main.coffee

```diff
@@ -52,8 +53,17 @@ class MPHistory.Main extends React.Component
     .done (data) =>
       return if _.isEmpty data.events.data
 
-      newEvents = _.concat @state.events, data.events.data
-      lastEvent = _.last(newEvents)
+      lastIndex = _.findLastIndex @state.events, (o) => o.id == @state.since + 1
```

Comment:
\<something about the next event id is not necessarily going to be `previous id + 1`\>

```coffeescript
  ...
  data:
    since: @minNextEventId() # wtf is even "since"
  ...

  previousEvents = _.filter @state.events, (e) =>
    e.id < data.events.data[0].id

  newEvents = _.concat previousEvents, data.events.data
...

minNextEventId: =>
  lastOngoingGame = ._findLast @state.events, (e) =>
    e.game? && !e.game.date.end_time?

  if lastOngoingGame?
    lastOngoingGame?.id - 1
  else
    _.last(@state.events)?.id ? 0
```